### DISCUSSION TITLE
Code do not run.

### DISCUSSION THREAD

**Discussion by chyasirabbasjutt** [cu-esiil-edu/2024-data-short-course#105] [Q&A] [answered]
I tried many times but cannot solve the problem. 
![image](https://github.com/cu-esiil-edu/2024-data-short-course/assets/166647114/c8a45dc2-3f25-487a-b5f7-3b88f6d91f94)
Please  Help me @nquarder @eculler 
How to resolve the issue?

**Comment by nquarder** [comment, marked as answer]
Hi @chyasirabbasjutt and good to hear from you my friend. Do you have geopandas (as gpd) imported at the top of your notebook? -NQ

**Reply by chyasirabbasjutt** [reply]
Thank you sir @nquarder.
How to resolve the issue.
![image](https://github.com/cu-esiil-edu/2024-data-short-course/assets/166647114/fba74005-a72f-4379-ae74-02daafaf9b5a)

# Reproject fire data 
fires_2021_4326 = fires_2021.to_crs('epsg:4326')
fires_2021_4326
I can not understand these words 'epsg:4326'.

**Reply by nquarder** [reply]
Okay @chyasirabbasjutt a couple things:

1.) do you have matplotlib.pyplot as plt imported? If not add the following to your import cell at the top of the notebook and re-run that cell:

`import matplotlib.pyplot as plt`


2.) epsg:4326 is a coordinate reference system. If you put it into Google you will see that it corresponds to :
https://epsg.io/4326

In this step we are taking the fire data from its original coordinate reference system and reprojecting it into EPSG: 4326. That way the fire data, the boundaries, and the NDVI data will all be in the same coordinate reference system and you'll be able to plot them all on the same axis without having your data look distorted. 

Give this a try @chyasirabbasjutt and let us know how it goes! 👍 -NQ

**Comment by chyasirabbasjutt** [comment]
Thanks Sir @nquarder

**Comment by chyasirabbasjutt** [comment]
Today, My code space is stopped automatically after 180 minutes. Is there any solution to solve the problem? @eculler @nquarder 
I worked on the whole shapefile of the Punjab Province. 
I rerun the code. 
![image](https://github.com/cu-esiil-edu/2024-data-short-course/assets/166647114/a1f19a6d-f7f2-4149-9d1a-1aab4066d918)

**Reply by nquarder** [reply]
@chyasirabbasjutt do you see the bar at the top of the screen asking you to enter your USERNAME for NASA Earthdata? You need to enter your USERNAME and PASSWORD for Earthdata. Then you will get an email once your request has been sent, and another email once your request has completed. Be sure you have credentials with Earthdata before you start this assignment (https://urs.earthdata.nasa.gov/users/new). Try this and it should help! -NQ

**Reply by chyasirabbasjutt** [reply]
ok, sir. @nquarder
